app/scripts/modules/core/src/domain/IExecutionStage.ts

```diff
@@ -62,10 +62,10 @@ export interface IExecutionStageSummary extends IOrchestratedItem {
   group?: string;
   groupStages?: IExecutionStageSummary[];
   id: string;
-  inSuspendedExecutionWindow?: boolean;
+  suspendedStageTypes: Set<string>;
   index: number;
-  labelComponent?: React.ComponentClass<IExecutionStageLabelComponentProps>;
-  markerIcon?: React.ComponentClass<{ stage: IExecutionStageSummary }>;
+  labelComponent?: React.ComponentType<IExecutionStageLabelComponentProps>;
+  markerIcon?: React.ComponentType<{ stage: IExecutionStageSummary }>;
```

Comment:
some stuff — including these type changes — comes from the first commit where I tried to re-use some wait stage code before going down the execution window route. I removed the gluey bits from that approach but left any generic refactors like widening these types to support functional components.